### PR TITLE
Baf 877/send statuses when moving

### PR DESCRIPTION
**Pull request by MarioIvancik**
Closes:
https://youtrack.bringauto.com/issue/BAF-881/Modules-not-checking-device-type-of-command
https://youtrack.bringauto.com/issue/BAF-877/Mission-module-only-sending-status-on-state-change

### CODE REVIEW

File: include/bringauto/modules/mission_module/devices/AutonomyDevice.hpp [hunk 1]
```diff
@@ -90,6 +93,10 @@ class AutonomyDevice {
 	static MissionModule::AutonomyCommand generateCommand(std::vector<MissionModule::Station> stops, std::string route,
 														  MissionModule::AutonomyCommand::Action action);
 
+	/**
+	 * @brief Map of last sent status timestamps for each device type
+	 */
+	inline static std::map<unsigned int, std::chrono::milliseconds> last_sent_status_timestamps_ {};
```
Comment (koudis) [inline include/bringauto/modules/mission_module/devices/AutonomyDevice.hpp:99]: Do not use INLIE static in header files!

Comment (koudis) [reply include/bringauto/modules/mission_module/devices/AutonomyDevice.hpp:99]: Explain what is firrefence betwen inline static in header file and inline static in c/cpp files.

File: source/bringauto/modules/mission_module/devices/AutonomyDevice.cpp [hunk 2]
```diff
@@ -2,25 +2,36 @@
...
 
 
 
 namespace bringauto::modules::mission_module::devices {
 
-int AutonomyDevice::send_status_condition(const struct buffer current_status, const struct buffer new_status) {
+/**
+ * @brief Map of last sent status timestamps for each device type
+ */
+static std::map<unsigned int, std::chrono::milliseconds> last_sent_status_timestamps_ {};
```
Comment (koudis) [inline source/bringauto/modules/mission_module/devices/AutonomyDevice.cpp:16]: PLease let the symbol as part of the class.

File: include/bringauto/modules/mission_module/devices/AutonomyDevice.hpp (hunk 1 above)
Comment (MarioIvancik) [reply include/bringauto/modules/mission_module/devices/AutonomyDevice.hpp]: is no longer inline

File: source/bringauto/modules/mission_module/devices/AutonomyDevice.cpp (hunk 2 above)
Comment (MarioIvancik) [reply source/bringauto/modules/mission_module/devices/AutonomyDevice.cpp]: declaration moved to header

**Review by koudis — APPROVED** [review]

**Pull request merged**